src/physics/mod.rs

```diff
@@ -4,12 +4,148 @@ pub mod rapier;
 #[cfg(feature = "avian")]
 pub mod avian;
 
-use bevy::prelude::*;
+use std::sync::Arc;
+
+use bevy::{ecs::system::EntityCommands, prelude::*};
 use bevy_ecs_tilemap::prelude::*;
 use tiled::{ObjectData, ObjectLayerData};
 
 use crate::prelude::*;
 
+/// Trait defining a generic way of handling colliders
+/// across different physics backends.
+pub trait HandleColliders {
+    /// Load shapes from an object layer as physics colliders.
+    ///
+    /// By default `bevy_ecs_tiled` will only process object layers
+    /// named in `collision_layer_names` in `TiledMapSettings`,
+    /// and tileset collision shapes named in `collision_object_names`.
+    ///
+    /// Collision layer names are case-insensitive and leading/trailing
+    /// whitespace is stripped out.
+    fn insert_colliders_from_shapes(
+        &self,
+        parent_entity: Entity,
+        map_type: &TilemapType,
+        grid_size: Option<&TilemapGridSize>,
+        object_data: &ObjectData,
+    ) -> EntityCommands;
+}
+
+#[derive(Clone, Resource)]
+pub enum PhysicsBackend {
+    Rapier,
+    Avian,
+    Custom(Arc<Box<dyn HandleColliders + Send + Sync>>),
+}
+
+impl PhysicsBackend {
+    pub fn insert_object_colliders(
+        &self,
+        commands: &mut Commands,
+        object_entity: Entity,
+        map_type: &TilemapType,
+        object_data: &ObjectData,
+        collider_callback: ColliderCallback,
+    ) {
+        match self {
+            #[cfg(feature = "rapier")]
+            PhysicsBackend::Rapier => {
+                let mut e = rapier::insert_rapier_colliders_from_shapes(
+                    commands,
+                    object_entity,
+                    map_type,
+                    None,
+                    object_data,
+                    collider_callback,
+                );
+
+                collider_callback(&mut e);
+            }
+            #[cfg(feature = "avian")]
+            PhysicsBackend::Avian => {
+                let e = avian::insert_avian_colliders_from_shapes(
+                    commands,
+                    object_entity,
+                    map_type,
+                    None,
+                    object_data,
+                );
+
+                let mut e = e.expect("failure creating entity for Avian collider");
```

Comment:
Why use `expect()` ? Why only for Avian ?
I believe you should remove it: if for some reason a collider cannot be added, we don't want the whole application to crash. It could be a "normal" behaviour to return `None` in case nothing is added.

On a side note, I would prefer this kind of structure to avoid repeating the call to `collider_callback`:

```rust
let e = match self {
  PhysicsBackend::Rapier => {
    rapier::insert_rapier_colliders_from_shapes( ... )
  }
  PhysicsBackend::Rapier => {
    avian::insert_avian_colliders_from_shapes( ... )
  }
  // ...
}

if let Some(entity_commands) = e {
  collider_callback(&mut entity_commands);
}
```